lib/foreman_inventory_upload/generators/slice.rb

```diff
@@ -97,13 +97,24 @@ def report_system_profile(host)
             end.join(', '))
           end
         end
-        @stream.array_field('installed_packages', :last) do
+        @stream.array_field('installed_packages') do
           first = true
           host.installed_packages.each do |package|
             @stream.raw("#{first ? '' : ', '}#{@stream.stringify_value(package.nvra)}")
             first = false
           end
         end
+        @stream.array_field('facts', :last) do
+          @stream.object do
+            @stream.simple_field('namespace', 'satellite')
+            @stream.object_field('facts', :last) do
+              if defined?(ForemanThemeSatellite)
+                @stream.simple_field('satellite_version', ForemanThemeSatellite::SATELLITE_VERSION)
+              end
+              @stream.simple_field('satellite_instance_id', Foreman.instance_id, :last)
```

Comment:
```suggestion
              @stream.simple_field('satellite_instance_id', Foreman&.instance_id, :last)
```

I intend using this plugin in versions before 1.24, where `instance_id` is not there yet. Can we add defensive code here?

Reply:
Do we want to keep the key even if we don't know the value?

Reply:
`simple_field` knows to deal with that. It will not write the field at all.